src/main/java/com/rultor/profiles/Profiles.java

```diff
@@ -73,33 +78,107 @@ public Profile fetch(final Talk talk) throws IOException {
+     *  fork branch. We need real `git merge` result here. See some details
+     *  here: https://github.com/yegor256/rultor/pull/1064/files#r56796959
+     */
+    public Profile validated(final Profile master, final Profile merged)
+        throws IOException {
+        final String commanders = "commanders";
+        final String[][] security = {
+            {"architect"},
+            {Profiles.MERGE, commanders},
+            {"deploy", commanders},
+            {"release", commanders},
+        };
+        for (final String[] section : security) {
+            if (!Arrays.equals(
+                Profiles.section(master, section),
+                Profiles.section(merged, section)
+            )) {
+                throw new Profile.ConfigException(
+                    String.format(
+                        "You cannot change `%s` section for security reasons",
+                        Arrays.toString(section)
+                    )
+                );
+            }
+        }
+        return merged;
+    }
+
+    /**
+     * Get section content.
+     * @param profile Profile
+     * @param section Section of profile
+     * @return Content of section as array of strings
+     * @throws IOException If fails
+     */
+    private static String[] section(final Profile profile,
+        final String... section)
+        throws IOException {
+        final StringBuilder path = new StringBuilder(Tv.HUNDRED);
+        for (final String element : section) {
+            path.append(String.format("/entry[@key='%s']", element));
+        }
+        final List<String> result = profile.read().xpath(
+            String.format("/%s/item/text()", path.toString())
+        );
+        return result.toArray(new String[result.size()]);
+    }
+
+    /**
      * Fetch a profile from an XML.
      * @param xml The XML
      * @return Profile found
+     * @throws IOException If fails
      */
-    private static Profile fetch(final XML xml) {
+    private Profile fetch(final XML xml) throws IOException {
         final Profile profile;
         final List<String> type = xml.xpath("//request/type/text()");
-        if (type.isEmpty() || !"merge".equals(type.get(0))) {
+        if (type.isEmpty() || !Profiles.MERGE.equals(type.get(0))) {
             profile = new GithubProfile(
                 new TalkIssues(Profiles.github(), xml).get().repo()
             );
         } else {
-            profile = new GithubProfile(
-                Profiles.github().repos().get(
-                    new Coordinates.Simple(
-                        xml.xpath(
-                            "//request/args/arg[@name='fork']/text()"
-                        ).get(0)
-                    )
-                ),
+            final Profile master = new GithubProfile(
+                    new TalkIssues(Profiles.github(), xml).get().repo()
+            );
+            profile = this.merged(
+                master,
```

Comment:
@xupyprmv the variable `master` is redundant here, please inline this real quick :)